tox.ini

```diff
@@ -1,27 +1,64 @@
 [tox]
-envlist = py38
+env_list =  # Environments to run when called with no parameters.
+minversion = 4.3.5
+# Tox will use these requirements to bootstrap a venv if necessary.
+# tox-igore-env-name-mismatch allows us to have one virtualenv for all linting.
+# By setting requirements here, we make this INI file compatible with older
+# versions of tox. Tox >= 3.8 will automatically provision the version provided
+# inside of a virtual environment, so users of Ubuntu >= focal can simply
+# install tox from apt. Older than that, the user gets an upgrade warning.
+requires =
+    # renovate: datasource=pypi
+    tox==4.4.7
+    # renovate: datasource=pypi
+    tox-ignore-env-name-mismatch==0.2.0.post2
+# Allow tox to access the user's $TMPDIR environment variable if set.
+# This workaround is required to avoid circular dependencies for TMPDIR,
+# since tox will otherwise attempt to use the environment's TMPDIR variable.
+user_tmp_dir = {env:TMPDIR}
 
-[testenv]
-setenv =
-    PYTHONPATH = {toxinidir}
-deps =
-    -r{toxinidir}/requirements-dev.txt
-    -r{toxinidir}/requirements.txt
-commands =
-    pip install -U pip
-    pytest --basetemp={envtmpdir}
+[testenv]  # Default config for all environments. Overridable in each env.
+# We have many tests that create temporary files. Unless the user has set a
+# TMPDIR, this will prefer putting those temp files in $XDG_RUNTIME_DIR,
+# which will speed up those tests since they'll run on a ramdisk.
+env_tmp_dir = {user_tmp_dir:{env:XDG_RUNTIME_DIR:{work_dir}}}/tox_tmp/{env_name}
+set_env =
+    TMPDIR={env_tmp_dir}
+    COVERAGE_FILE={env_tmp_dir}/.coverage_{env_name}
 
-[testenv:docs]
-commands = make docs
+[docs]  # Sphinx documentation configuration
+deps = -r requirements-jammy.txt
+extras = doc
+package = editable
+no_package = true
+env_dir = {work_dir}/docs
+runner = ignore_env_name_mismatch
+allowlist_externals = bash
+commands_pre =
+    bash -c 'if [[ ! -e docs ]];then echo "No docs directory. Run `tox run -e sphinx-quickstart` to create one.";return 1;fi'
+    bash -c "rm -f {tox_root}/docs/common/craft-parts"
+    bash -c "ln -fs $(python -c 'import craft_parts_docs;print(craft_parts_docs.__file__.rsplit(\"/\", maxsplit=1)[0])') {tox_root}/docs/common/craft-parts"
 
-[testenv:lint]
-commands = make lint
+[testenv:link-docs-pkg]
+description =
+    Use a local (editable) dependency package for documentation rather than the version in requirements.
```

Comment:
What's the use case here?  Linking to a particular craft-parts branch for informal testing?